tests/integration/test_charm.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+
+import logging
+import pathlib
+import time
+
+import pytest
+import requests
+import yaml
+from pytest_operator import plugin as pytest_plugin
+
+logger = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(pathlib.Path("./metadata.yaml").read_text())
+APP_NAME = METADATA["name"]
+POSTGRESQL_CHARM = "postgresql-k8s"
+NGINX_INGRESS_CHARM = "nginx-ingress-integrator"
+
+
+def check_waltz_connection(url, headers=None):
+    """Tests the connection to the given URL.
+
+    Returns True if the response status code is 200.
+    """
+    # When the Waltz container starts, it is running Liquibase before starting the service itself.
+    # We should retry a few times, to give it time to start the Waltz service.
+    # Similarly, when the Ingress Route is being established, it takes a few seconds to take
+    # effect.
+    for i in range(30):
+        logger.info("Trying to access Waltz...")
+        try:
+            response = requests.get(url, headers=headers)
+            if response.status_code == 200:
+                return True
+        except Exception as ex:
+            logger.info(ex)
+
+        time.sleep(2)
```

Comment:
This retry logic feels a bit naive, could we replace this with tenacity please? Something like [this](https://github.com/canonical/mongodb-operator/blob/5d5e7c1ba3b4a1a3a5c25d170bfc88c819386290/tests/integration/test_charm.py#L207)

Reply:
Done